app/controllers/users_controller.rb

```diff
@@ -18,14 +18,26 @@ def new
   def create
     @user = User.new(user_params)
     if @user.save
-      session[:user_id] = @user.id
-      flash[:notice] = "Welcome #{@user.username}! You have successfully signed up."
-      redirect_to articles_path
+      UserMailer.welcome_email(@user).deliver_now
+      flash[:notice] = "Please check your emails to confirm your sign up"
+      redirect_to root_path
     else
+      flash[:error] = "Ooooppss, something went wrong!"
       render 'new'
     end
   end
 
+  def confirm_email
+    user = User.find_by_confirm_token(params[:id])
+    if user
+      user.email_activate 
```

Comment:
wir können auch einfach schreiben:
```suggestion
      user.update_attribute(:email_confirmed, true)
```
Dann sparen wir uns die Methode im Model...

Reply:
Also, ich habe herum gespielt... Ich möchte die Methode `email_activate` behalten, weil ich möchte, dass der Link abläuft, nachdem der BenutzerIn ihn in der Mail angeklickt hat.

Reply:
Ich möchte auch auf diesen Grund  auch `self.confirm_token = nil` da lassen

Reply:
Also es passiert auch nichts, wenn der Link mehrmals geklickt wird, Aber es stört auch nicht, den Token zu entfernen.
Meistens hat man noch ein Ablaufdatum am Token, dann wird es klarer, dass die Methode benötigt wird.